pygeos/tests/test_creation.py

```diff
@@ -44,9 +44,12 @@ def test_points_from_xyz():
 
 
 def test_points_invalid_ndim():
-    with pytest.raises(pygeos.GEOSException):
+    with pytest.raises(ValueError, match="dimension should be 2 or 3, got 4"):
         pygeos.points([0, 1, 2, 3])
```

Comment:
This change is probably causing the failures. When I changed this, I also updated ufuncs.c to raise this error ourselves: https://github.com/shapely/shapely/commit/5a8de7831371fdde10c7d97064e066fb4e55eb4d#diff-312b6c26b151e9c8928057e9a72c074d8b3fd6fdc8dea09800efb753b5612b98R2459

Note that this specific PR was only needed for GEOS main, not for GEOS <= 3.11.